### PR TITLE
Dynamic class

### PR DESCRIPTION
**Pull request by MartinMystikJonas**
RelatedFilesResolver will fail when it found `new $class()`

### CODE REVIEW

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/ResolvePresenter.php [hunk 1]
```diff
@@ -94,4 +94,10 @@ public function actionSendTemplateUnresolvable(): void
         $this->template->send = 'send';
         $this->sendTemplate(new DefaultTemplate(new Engine()));
     }
+
+    public function dynamicClassName()
+    {
+        $class = uniqid();
```
Comment (lulco) [inline tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/ResolvePresenter.php:100]: What about some string representing existing class name? It should work.

Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/ResolvePresenter.php:100]: I think we could use value resolver if it is expression

Comment (lulco) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/ResolvePresenter.php:100]: You have scope here, isn't it better?

Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/ResolvePresenter.php:100]: Value resolver uses scope internally in cases where it is useful

Comment (lulco) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/Fixtures/ResolvePresenter.php:100]: 👍

**Review by lulco — APPROVED** [review]

**Comment by MartinMystikJonas** [comment]
I added resolveStrings to ValueResolver because in almost all palces we need just strings and we were filtering non+string values out repeatedly. Ok?

File: src/Resolver/ValueResolver/ValueResolver.php [hunk 2]
```diff
@@ -104,4 +104,16 @@ public function resolve(Expr $expr, Scope $scope, $unknownValuePlaceholder = nul
             return null;
         }
     }
+
+    /**
+     * @return string[]|null
+     */
+    public function resolveStrings(Expr $expr, Scope $scope)
```
Comment (lulco) [inline src/Resolver/ValueResolver/ValueResolver.php]: Add return type ?array

**Review by lulco — APPROVED** [review]

**Pull request merged**